dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java

```diff
@@ -698,4 +709,35 @@ private static void usage() {
             System.out.println("     -h Shows this text");
         }
     }
+
+    /**
+     * Method to cross references handles of items that have been deleted with OAI core docs of said handle.
+     * If a match is found update that document.
+     *
+     * @param solrServer
+     *            The configured solr server to commit to
+     * @param handle
+     *            The handle to cross reference
+     *
+     */
+    private void updateDocsWithDeletes(SolrClient solrServer, String handle) {
+        try {
+            SolrQuery params = new SolrQuery("item.handle:" + handle);
```

Comment:
this will only work if we still have the record in the oai core. We should support also the case where a full reindex is triggered. You can delete the existing record (if any) and create a new one with just the ids and the deleted flag

Reply:
@abollini Thanks for the feedback! I believe I addressed the improvements you brought up. You'll notice I added `XOAI.SolrSearchQuerySingleSupportNull()` , this was purely to help simplify `xoai.updateDocsWithDeletes()` without changing the design of `DSpaceSolrSearch.querySingle()`.  But I'm open to using the pattern here: https://github.com/DSpace/DSpace/blob/main/dspace-oai/src/main/java/org/dspace/xoai/services/impl/xoai/DSpaceItemSolrRepository.java#L63 if you would prefer.